pkg/standalone-utils/contracts/PermitRelayer.sol

```diff
@@ -0,0 +1,126 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+pragma solidity ^0.7.0;
+pragma experimental ABIEncoderV2;
+
+import "@balancer-labs/v2-vault/contracts/interfaces/IVault.sol";
+import "./base/Multicall.sol";
+import "./base/VaultPermit.sol";
+import "./interfaces/IPermit.sol";
+import "./interfaces/IPermitDAI.sol";
+
+/**
+ * @title PermitRelayer
+ * @notice Allows users to atomically call permit on multiple tokens and then interact with the Balancer Vault
+ * @dev This relayer is extremely powerful and has full control over users' assets should they authorise it.
+ * For this reason, users should only perform signature-based approvals to allow it to perform only a single action.
```

Comment:
I'd frame this as a vault multicall contract that lets the caller perform multiple Vault actions (out of a subset) in sequence, along with token approvals, allowing for signature-based single transaction interactions (including onboarding!).